presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/util/IcebergAvroDataConversion.java

```diff
@@ -0,0 +1,348 @@
+        private Schema icebergSchema;
+        private int position;
+        private int positionCount;
+
+        public RecordIterator(Page page, List<Type> types, Schema icebergSchema)
+        {
+            requireNonNull(page, "page is null");
+            this.types = ImmutableList.copyOf(requireNonNull(types, "types is null"));
+            this.icebergSchema = requireNonNull(icebergSchema, "icebergSchema is null");
+            checkArgument(page.getChannelCount() == types.size(), "the page's channel count must be equal to the size of types");
+            checkArgument(types.size() == icebergSchema.columns().size(), "the size of types must be equal to the number of columns in icebergSchema");
+            icebergTypes = icebergSchema.columns().stream()
+                    .map(Types.NestedField::type)
+                    .collect(toImmutableList());
+            columnBlocks = IntStream.range(0, types.size())
+                    .mapToObj(page::getBlock)
+                    .collect(toImmutableList());
+            positionCount = page.getPositionCount();
+        }
+
+        @Override
+        public boolean hasNext()
+        {
+            return position < positionCount;
+        }
+
+        @Override
+        public Record next()
+        {
+            Record record = GenericRecord.create(icebergSchema);
+            for (int channel = 0; channel < types.size(); channel++) {
+                Object element = toIcebergAvroObject(types.get(channel), icebergTypes.get(channel), columnBlocks.get(channel), position);
+                record.set(channel, element);
+            }
+            position++;
+            return record;
+        }
+    }
+
+    public static Object toIcebergAvroObject(Type type, org.apache.iceberg.types.Type icebergType, Block block, int position)
+    {
+        if (block.isNull(position)) {
+            return null;
+        }
+        if (type.equals(BOOLEAN)) {
+            return type.getBoolean(block, position);
+        }
+        if (type.equals(BIGINT)) {
+            return type.getLong(block, position);
+        }
+        if (type.equals(INTEGER)) {
+            return toIntExact(type.getLong(block, position));
+        }
+        if (type.equals(REAL)) {
+            return intBitsToFloat((int) type.getLong(block, position));
+        }
+        if (type.equals(DOUBLE)) {
+            return type.getDouble(block, position);
+        }
+        if (type instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) type;
+            BigInteger unscaledValue;
+            if (decimalType.isShort()) {
+                unscaledValue = BigInteger.valueOf(decimalType.getLong(block, position));
+            }
+            else {
+                unscaledValue = Decimals.decodeUnscaledValue(decimalType.getSlice(block, position));
+            }
+            return new BigDecimal(unscaledValue, decimalType.getScale());
```

Comment:
should we use `new BigDecimal(unscaledValue, decimalType.getScale(), type.getPrecision())` ? Or may be just `Decimals#readBigDecimal`